syntax = "proto3";

package masterworker;

service Worker {
    rpc ExecuteMapJob (MapJobRequest) returns (MapJobReply) {}
    rpc ExecuteReduceJob (ReduceJobRequest) returns (ReduceJobReply) {}
}

message MapJobRequest {
    message FileOffset {
        string file = 1;
        int32 start = 2;
        int32 stop = 3;
    }

    repeated FileOffset offsets = 1;
}


message MapJobReply {
    bool success = 1;
}

message ReduceJobRequest {
    string key = 1;
}

message ReduceJobReply {
    bool success = 1;
}